code/datums/outfits/outfit.dm

```diff
@@ -272,16 +295,18 @@
 		if(H.mind && H.mind.initial_account)
 			C.associated_account_number = H.mind.initial_account.account_number
 
-/datum/outfit/proc/imprint_pda(mob/living/carbon/human/H, obj/item/device/pda/PDA)
-	var/obj/item/card/id/C = PDA.id
-	PDA.owner = H.real_name
-	if(istype(PDA) && istype(C))
-		PDA.ownjob = C.assignment
-		PDA.ownrank = C.rank
-	else //As a fallback if the id isnt inside of the PDA
-		PDA.ownjob = get_id_assignment(H)
-		PDA.ownrank = get_id_rank(H)
-	PDA.name = "PDA-[H.real_name] ([PDA.ownjob])"
+/datum/outfit/proc/imprint_pda(mob/living/carbon/human/H, obj/item/modular_computer/PDA)
+	PDA.name = "PDA-[H.real_name] ([get_id_assignment(H)])"
+	PDA.update_icon()
+
+/datum/outfit/proc/register_pda(obj/item/modular_computer/P, obj/item/card/id/I)
+	P.card_slot.insert_id(I)
+	if(!P.hidden)
+		return
+	if(P.card_slot && P.card_slot.stored_card)
```

Comment:
```suggestion
	if(P.card_slot.stored_card)
```

If we do not have a card slot, we already runtimed on line 303. This check should be before line 303.